platform/features/layout/src/LayoutController.js

```diff
@@ -293,9 +347,101 @@ define(
          * view configuration.
          */
         LayoutController.prototype.endDrag = function () {
+            this.frameMoved = true;
+
+            setTimeout(function () {
+                this.frameMoved = false;
+            }.bind(this), 0);
+
             this.endDragInScope();
         };
 
+        /**
+         * Check if the object is currently selected.
+         *
+         * @param {string} obj the object to check for selection
+         * @returns {boolean} true if selected, otherwise false
+         */
+        LayoutController.prototype.selected = function (obj) {
+            return !!this.selectedId && this.selectedId === obj.getId();
+        };
+
+        /**
+         * Set the active user selection in this view.
+         *
+         * @param event the mouse event
+         * @param {string} id the object id
+         */
+        LayoutController.prototype.select = function (event, id) {
+            if (event) {
+                event.stopPropagation();
+                if (this.selection) {
+                    event.preventDefault();
+                }
+            }
+
+            this.selectedId = id;
+
+            var selectedObj = {};
+            selectedObj[this.frames[id] ? 'hideFrame' : 'showFrame'] = this.toggleFrame.bind(this, id);
+
+            if (this.selection) {
+                this.selection.select(selectedObj);
+            }
+        };
+
+        /**
+         * Callback to show/hide the object frame.
+         *
+         * @param {string} id the object id
+         * @private
+         */
+        LayoutController.prototype.toggleFrame = function (id) {
+            var configuration = this.$scope.configuration;
+
+            if (!configuration.panels[id]) {
+                configuration.panels[id] = {};
+            }
+
+            this.frames[id] = configuration.panels[id].hasFrame = !this.frames[id];
+            this.select(undefined, id); // reselect so toolbar updates
+        };
+
+        /**
+         * Clear the current user selection.
+         */
+        LayoutController.prototype.clearSelection = function () {
+            // Keep the selection if the frame is moved.
+            if (this.frameMoved) {
+                this.frameMoved = false;
```

Comment:
Good catch! I removed that line and updated the test to tick the clock before calling clearSelection the second time. Also, changed the it text to ""prevents clearing selection while drag is in progress".

Keeping "clears selection after moving/resizing" which tests clearSelection for the case where the mouse position did not cause clearSelection to be called. Let me know if that's unnecessary.